nearcore/src/runtime/mod.rs

```diff
@@ -711,7 +712,13 @@ impl RuntimeAdapter for NightshadeRuntime {
         pool_iterator: &mut dyn PoolIterator,
         chain_validate: &mut dyn FnMut(&SignedTransaction) -> bool,
         current_protocol_version: ProtocolVersion,
+        time_limit: Option<Duration>,
```

Comment:
It's an `Option` in the function because someone might want to use this function without a time limit (e.g for tests).
For the main config I didn't see any real reason to not use a time limit, so I didn't make it optional.
I can make it an `Option` in the config as well, just to keep things consistent.